src/driver/kvm.rs

```diff
@@ -1,32 +1,58 @@
+use std::collections::HashMap;
+use std::convert::TryInto;
 use std::error::Error;
 
-use kvmi::{KVMi, KVMiEventReply, KVMiEventType};
+use kvmi::{KVMi, KVMiCr, KVMiEvent, KVMiEventReply, KVMiEventType, KVMiInterceptType};
 
-use crate::api::{DriverType, Introspectable, Registers, X86Registers};
+use crate::api::{
+    CrType, DriverType, Event, EventReplyType, EventType, InterceptType, Introspectable, Registers,
+    X86Registers,
+};
 
 // unit struct
 #[derive(Debug)]
 pub struct Kvm {
     kvmi: KVMi,
     expect_pause_ev: u32,
+    // VCPU -> KVMiEvent
+    map_events: HashMap<u16, KVMiEvent>,
```

Comment:
nitpick: From what I understand you want to register a single event per vcpu, right? Isn't a hash map a little bit of an overkill for this purpose? I would just use a vector instead. You don't necessarily need to change this, I mainly want to know if I have missed something. If so, please tell me ;)

Reply:
Yes, it's totally overkill.

I just wanted to store the KVMEvent in the KVM driver, and associate its VCPU id to find it back for the response.

A vector is good also.
 
And i'm not storing singlestep events but general KVMevents.

Reply:
The last version uses a Vector instead.